src/other/index.jl

```diff
@@ -113,8 +113,13 @@ Base.getindex(x::AbstractIndex, idx::Real) = Int(idx)
 Base.getindex(x::AbstractIndex, idx::AbstractVector{Union{Bool, Missing}}) =
     getindex(x, collect(Missings.replace(idx, false)))
 Base.getindex(x::AbstractIndex, idx::AbstractVector{Bool}) = find(idx)
-Base.getindex(x::AbstractIndex, idx::AbstractVector{T}) where {T >: Missing} =
-    getindex(x, collect(skipmissing(idx)))
+function Base.getindex(x::AbstractIndex, idx::AbstractVector{T}) where {T >: Missing}
+    idx2 = collect(skipmissing(idx)) # idx can be Any and we have to handle it
```

Comment:
Actually I think there would be a cleaner and more efficient way to do this: 
```julia
Int[i isa Real ? i : x[i] for i in skipmissing(idx)]
```

Reply:
After thinking about it I propose an implementation which disallows mixing `Real` and `Symbol` indexing, so that something like this `df[[1, :x2, missing]]` will throw an error (in the old implementation it would go through).